### ISSUE TITLE
Add information to help locate the error in the template

### ISSUE THREAD

**Issue by lopuhin** [elapouya/python-docx-template#378]
When there is an error in the template, it's often hard to locate the exact location of the error. Vanilla jinja2 provides the line number in the template, and then it's normally trivial to fix the error.

It would be great to have come context information. E.g. text before / after the error would already be a huge help.

**Comment by anitatea** [comment]
Would love to see this!

Something using the XML to pin point where the error occurred / rendered stopped at within the template. Beginner in python here - have been using your library for a few months to create reports. 

I'm trying to build this "debug"-ish feature myself with little success @elapouya - if you have any advice or notes if you had started to work on this previously, I'd love to try to help or test!

**Comment by ccvivien** [comment]
I would also love to see this. Right now, it's a bit difficult to debug a template.

**Comment by ccvivien** [comment]
After reading the source code, it's already implemented.

if you catch `jinja2.exceptions.TemplateError` you have access to 2 attributes : `lineno` and `docx_context`

eg: 
```
try:
    doc.render(ctx, jinja_env)
except jinja2.exceptions.TemplateError as e:
    print(list(e.docx_context))
    print(e.lineno)
```

**Comment by ThbtSprt** [comment]
> ```
> try:
>     doc.render(ctx, jinja_env)
> except jinja2.exceptions.TemplateError as e:
>     print(list(e.docx_context))
>     print(e.lineno)
> ```

@ccvivien Good idea but it doesn't helped me with `jinja2.exceptions.TemplateSyntaxError`  :
the ``e.docx_context`` appears to stand in only one line, even if the template is composed of multiple paragraphs, so it seems that the ``e.lineno`` always refers to line n°1...

**Comment by ccvivien** [comment]
Weird. Because if you look here https://github.com/pallets/jinja/blob/417f822196f66155e8c121e5229cc12a6b02ce14/src/jinja2/exceptions.py

you can see that `TemplateSyntaxError` is the only error type (and its child) that includes `lineno`.
So context and line number should definitely be available.
For example, if you have an `UndefinedError`, no context and line number will be available.

**Comment by ThbtSprt** [comment]
You're right, the line number is available even for `TemplateSyntaxError`.

My problem was that the `*.docx_context` had only one line because the text was included in a table. 

Outside a table, it works perfectly, thanks.
